src/commands/android/subcommands/common.ts

```diff
@@ -74,3 +76,81 @@ export function showMissingBinaryHelp(binaryName: SdkBinary) {
   Logger.log(`Run: ${colors.cyan('npx @nightwatch/mobile-helper android --standalone')} to setup missing requirements.`);
   Logger.log(`(Remove the ${colors.gray('--standalone')} flag from the above command if setting up for testing.)\n`);
 }
+
+export function showHelp(subcommand: string) {
+  const subcmd = AVAILABLE_SUBCOMMANDS[subcommand];
+
+  Logger.log(`Usage: ${colors.cyan(`npx @nightwatch/mobile-helper android ${subcommand} [options]`)}\n`);
+  Logger.log(colors.yellow('Options:'));
+
+  const subcmdOptionsHelp = getSubcommandOptionsHelp(subcmd);
+  Logger.log(subcmdOptionsHelp);
+}
+
+export function verifyOptions(subcommand: string, options: Options): false | verifyOptionsResult {
+  const optionsPassed = Object.keys(options).filter(option => options[option] !== false);
+  const availableOptions = AVAILABLE_SUBCOMMANDS[subcommand].options;
+
+  const availableOptionsNames = availableOptions.map(option => option.name);
+
+  // Divide the optionsPassed array in two arrays: mainOptionsPassed and optionFlagsPassed.
+  // mainOptionsPassed contains the main option that is available for the subcommand.
+  // optionFlagsPassed contains the options with string or boolean values corresponding to the main option.
+
+  const mainOptionsPassed = optionsPassed.filter(option => availableOptionsNames.includes(option));
+  const optionFlagsPassed = optionsPassed.filter(option => !availableOptionsNames.includes(option));
+
+  if (mainOptionsPassed.length > 1) {
+    // A subcommand can only have one main option.
+    Logger.log(`${colors.red(`Too many options passed for subcommand ${subcommand}:`)} ${mainOptionsPassed.join(', ')}`);
+    showHelp(subcommand);
+
+    return false;
+  } else if (mainOptionsPassed.length === 0 && optionFlagsPassed.length) {
+    // If the main option is not present, then any other options present are invalid.
+    Logger.log(`${colors.red(`Unknown option(s) passed for subcommand ${subcommand}:`)} ${optionFlagsPassed.join(', ')}`);
+    showHelp(subcommand);
+
+    return false;
```

Comment:
Just asking, don't make any changes: wouldn't the above block misbehave for subcommands like `disconnect` which does not accept any main option but only the optionFlags?

Reply:
This function wasn't intended to be used for subcommands like disconnect. Since there is just one workflow for disconnect and no main option, so the `disconnect/index.ts` file itself is used for the disconnect script. Unlike other scripts where the `index.ts` file is used to verify options and facilitate default flow. In this case, there is just one thing we have to verify, that is, if there is any other option other than the valued flags then we will show error otherwise everything works fine.

So shall we just manually check this before calling the disconnect script (which I originally considered), or shall we made a minor addition to the `verifyOptions` to do this instead?

Reply:
Right now, I think updating `verifyOptions` looks a better idea to me.